terraso_backend/apps/graphql/schema/commons.py

```diff
@@ -37,19 +38,54 @@ class TerrasoConnection(Connection):
     class Meta:
         abstract = True
 
-    total_count = Int()
+    total_count = Int(required=True)
```

Comment:
Question: What is the mening of required here? is this an input value?

Reply:
`required` is shorthand for wrapping in `NonNull`: https://docs.graphene-python.org/en/latest/types/list-and-nonnull/#nonnull

it's an assertion that the field must resolve to a non-null value, or else the server will raise an error and the request will fail. in this case because it's resolving to a count of an iterable, it will always be true (unless the entire request fails, in which case some enclosing node of response tree will be null). making the assertion here: a) catches potential errors if we accidentally return null on the server, and b) simplifies client code since it no longer needs to handle potential null values on this field

Reply:
@shrouxm do you know if this validation executed on top of the requested fields in the query? or always before resolving the node fields?

Reply:
@josebui hmm i'm not sure what you mean, like does the validation happen at the very end before returning the result or does it happen right after the resolve function is called? in either case i'm not entirely sure, but maybe i can intentionally return `null` in the resolver and see what happens for our edification